app/Filament/Resources/UserResource/Pages/EditUser.php

```diff
@@ -54,6 +56,9 @@ protected function handleRecordUpdate(Model $record, array $data): Model
                 'email' => $data['email']
             ]);
 
+            $role = Role::find($data['role']);
+            User::find($record->id)->assignRole($role->name);
```

Comment:
@kolanaveen No need to find user again by `$record->id`. $record is already giving User model.

Reply:
Yes! But we need model instance to assign role when using spatie `assignRole` function, `$record` gives sql response